Data/Primitive/ByteArray.hs

```diff
@@ -223,17 +231,20 @@ foldrByteArray f z arr = go 0
       | otherwise                    = z
     sz = sizeOf (undefined :: a)
 
-fromListN :: Prim a => Int -> [a] -> ByteArray
-fromListN n ys = runST $ do
+byteArrayFromList :: Prim a => [a] -> ByteArray
+byteArrayFromList xs = byteArrayFromListN (length xs) xs
+
+byteArrayFromListN :: Prim a => Int -> [a] -> ByteArray
+byteArrayFromListN n ys = runST $ do
     marr <- newByteArray (n * sizeOf (head ys))
     let go !ix [] = if ix == n
           then return ()
-          else die "fromListN" "list length less than specified size"
+          else die "byteArrayFromListN" "list length less than specified size"
         go !ix (x : xs) = if ix < n
           then do
             writeByteArray marr ix x
             go (ix + 1) xs
-          else die "fromListN" "list length greater than specified size"
+          else die "byteArrayFromListN" "list length greater than specified size"
     go 0 ys
     unsafeFreezeByteArray marr
```

Comment:
At some point, this should probably use a `runByteArray` or `createByteArray` here.